.github/workflows/socat.yml

```diff
@@ -78,4 +78,4 @@ jobs:
         run: |
           export LD_LIBRARY_PATH=$GITHUB_WORKSPACE/build-dir/lib:$LD_LIBRARY_PATH
           export SHELL=/bin/bash
-          SOCAT=$GITHUB_WORKSPACE/socat-1.8.0.0/socat ./test.sh -t 0.5 --expect-fail 36,64,146,214,216,217,309,310,386,399,402,403,459,460,467,468,478,492,528,530
+          SOCAT=$GITHUB_WORKSPACE/socat-1.8.0.0/socat ./test.sh -t 0.5 --expect-fail 36,64,146,214,216,217,309,310,386,399,402,403,459,460,467,468,475,478,492,528,530
```

Comment:
Why are we expecting more failures with socat unit tests now?

Reply:
> socat.yml: in test 475, the test would send a SIGTERM after 3 seconds. We would continue to ignore this signal and continue to call `recvfrom`. Instead we should error out and give control back to the user.

Without this PR we would actually error out on `sendto`  with `EINVAL` due to passing in a bad peer parameter. With the peer handling refactoring we would no longer error out on `sendto` and be stuck in a `recvfrom` loop because we were ignoring the `EINTR` return. Now we return an error to the user.

Reply:
Sorry, it's not clear to me. Is it that this PR fixed handling `EINVAL` which was causing the test to succeed previously but in an unexpected way? Where it should have been waiting and returning back on a `SIGTERM` signal? And now with this change we are ignoring a different signal that should be handled for the socat test?

Reply:
Close. With the peer refactor we no longer trigger an invalid value error. When socat tries to kill the process it causes recvfrom to return with EINTR. Previously we would ignore this error and stall in an emdless recvfrom loop. Now I've changed the behaviour to return an error when the socket returns EINTR. Now we no longer stall but the socat test still fails. The test is trying to set the timeout directly on the socket. Our dtls code ignores this and overrides the timeout. I don't conaider this a failure so I've set the test to expect failure.